examples/language-modeling/run_mlm_wwm_flax.py

```diff
@@ -0,0 +1,315 @@
+    else:
+        config = CONFIG_MAPPING[model_args.model_type]()
+        logger.warning("You are instantiating a new config instance from scratch.")
+
+    if model_args.tokenizer_name:
+        tokenizer = AutoTokenizer.from_pretrained(
+            model_args.tokenizer_name, cache_dir=model_args.cache_dir, use_fast=model_args.use_fast_tokenizer
+        )
+    elif model_args.model_name_or_path:
+        tokenizer = AutoTokenizer.from_pretrained(
+            model_args.model_name_or_path, cache_dir=model_args.cache_dir, use_fast=model_args.use_fast_tokenizer
+        )
+    else:
+        raise ValueError(
+            "You are instantiating a new tokenizer from scratch. This is not supported by this script."
+            "You can do it from another script, save it, and load it from here, using --tokenizer_name."
+        )
+
+    # Preprocessing the datasets.
+    # First we tokenize all the texts.
+    if training_args.do_train:
+        column_names = datasets["train"].column_names
+    else:
+        column_names = datasets["validation"].column_names
+    text_column_name = "text" if "text" in column_names else column_names[0]
+
+    padding = "max_length" if data_args.pad_to_max_length else False
+
+    def tokenize_function(examples):
+        # Remove empty lines
+        examples["text"] = [line for line in examples["text"] if len(line) > 0 and not line.isspace()]
+        return tokenizer(examples["text"], padding=padding, truncation=True, pad_to_multiple_of=8, max_length=data_args.max_seq_length)
+
+    tokenized_datasets = datasets.map(
+        tokenize_function,
+        batched=True,
+        num_proc=data_args.preprocessing_num_workers,
+        remove_columns=[text_column_name],
+        load_from_cache_file=not data_args.overwrite_cache,
+    )
+
+    # Data collator
+    # This one will take care of randomly masking the tokens.
+    data_collator = DataCollatorForWholeWordMask(tokenizer=tokenizer, mlm_probability=data_args.mlm_probability)
+
+    # Initialize our training
+    rng = jax.random.PRNGKey(training_args.seed)
+
+    rng, init_rng = jax.random.split(rng)
+    model = FlaxBertForMaskedLM.from_pretrained("bert-base-cased")
+    model.init_head(init_rng)
+
+    # Setup optimizer
+    optimizer = Adam(
+        learning_rate=training_args.learning_rate,
+        weight_decay=training_args.weight_decay
+    ).create(model.params)
+
+    for epoch in track(range(int(training_args.num_train_epochs)), description="Training..."):
+        samples_idx = np.random.choice(len(tokenized_datasets["train"]), (training_args.train_batch_size, ))
+        samples = [tokenized_datasets["train"][idx.item()] for idx in samples_idx]
+        model_inputs = data_collator(samples)
+
+        # Model forward
+        # TODO: Remove this conversion by replacing the collator
+        model_inputs = {var_name: tensor.numpy() for var_name, tensor in model_inputs.items()}
+        loss, optimizer = training_step(optimizer, model_inputs)
```

Comment:
This is really cool!

Can you also easily have a learning rate schedule?

Reply:
Yap sure, didn't put it in at first, focusing on making things clear and almost a no-brainer 😄. 

Will look at it soon 👍

Reply:
Now included 👍